### ISSUE TITLE
[chai-a11y-axe] Promise exceeds timeout in build 

### ISSUE THREAD

**Issue by jorenbroekema** [open-wc/open-wc#1749]
```
Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

This happens every now and then for our accessibility tests in lion, and I cannot seem to reproduce it locally.

https://app.circleci.com/pipelines/github/ing-bank/lion/3691/workflows/9e5376f9-57f5-4907-b6a6-521824ff88e7/jobs/4215/steps

It's really inconsistent, so I'm not sure how to even start debugging this problem. I was wondering if you guys have any suggestions. I'm running the tests through web-test-runner.

Seems like the promise never resolves nor rejects, which makes me think it just gets stuck, but I'm not sure what it could get stuck on.. Any ideas?

**Comment by jarrodek** [comment]
You can set different timeout in your `karma.config.js` file by adding:

```
client: {
  mocha: {
     timeout: 10000,
  },
},
```

The a11y tests are processed by axe-core library and you could ask for help there. The testing configuration offered by open-wc only adds axe-core to Chai and passes the configuration to the library. I never had a problem with a11y testing (especially timeouts) but when such arise I usually extend the timeout a bit.

**Comment by jorenbroekema** [comment]
You think it could happen that the a11y analysis done by axe takes more than 2000ms on slower machines e.g. circleci docker image? I'll try to up the timeout a bit to see if that helps :) but 2000ms is quite a long time so I was wondering if it could be something else where it gets stuck indefinitely.

**Comment by Westbrook** [comment]
I've also found that on slower machines the `concurrency: Infinity,` setting in the config will flood the RAM of the boxes on Circle, which top out at 2 core 4 gb ram instances. We currently have the tuned all the way down at `concurrency: 1`, which helps reduce this, but there are still times when we hit the timeout anyways.

**Comment by jarrodek** [comment]
It's hard to say as it depends on the component. If it has a lot of DOM to scan then it might take a while. Remember that axe-core traverses the shadow root of each component that is added to your shadow root. This can grow very quickly without your intention.
I am using this module for a11y testing for months now and never had an issue with timeouts before. Maybe it's something else in your code? Try changing the timeout first and see what's happening. Maybe it's just a timeout on a slow machine.

**Comment by jpchase** [comment]
This is an old issue, but I've been seeing this problem of timeouts sporadically. After instrumenting the `chai-a11y-axe` code, the problem seems to be with the [dynamic import](https://github.com/open-wc/open-wc/blob/5aeaf35e01a2f15b6663d71102eeea3333e4c57d/packages/chai-a11y-axe/src/axe-import.js#L25) of the `axe-core` package. When the timeouts occur in a test, it looks like the import is called but the promise doesn't complete before the test times out. This can happen if there are multiple tests that use the `accessible` expect helper.

As a workaround, I used a global setup script to statically import `axe-core` before any tests run. This seems to ensure that `window.axe` is set, and then the dynamic import isn't needed.
